gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/util/ObservationUtil.java

```diff
@@ -0,0 +1,213 @@
+
+public class ObservationUtil {
+    private static final String VALUE_QUANTITY_EXTENSION = "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-CareConnect"
+        + "-ValueApproximation-1";
+    private static final String CODING_SYSTEM = "http://hl7.org/fhir/v2/0078";
+
+    private static final QuantityMapper QUANTITY_MAPPER = new QuantityMapper();
+
+    public static Quantity getValueQuantity(Object value, CV uncertaintyCode) {
+        if (isValidValueQuantity(value)) {
+            Quantity valueQuantity;
+            if (value instanceof PQ) {
+                valueQuantity = QUANTITY_MAPPER.mapQuantity((PQ) value);
+            } else {
+                valueQuantity = QUANTITY_MAPPER.mapQuantity((IVLPQ) value);
+            }
+
+            if (uncertaintyCode != null) {
+                valueQuantity.getExtension().add(new Extension()
+                    .setUrl(VALUE_QUANTITY_EXTENSION)
+                    .setValue(new BooleanType().setValue(true)));
+            }
+
+            return valueQuantity;
+        }
+
+        return null;
+    }
+
+    public static CodeableConcept getInterpretation(CV interpretationCode) {
+        if (interpretationCode != null) {
+            var interpretationCodeableConcept = new CodeableConcept();
+
+            var code = interpretationCode.getCode();
+
+            interpretationCodeableConcept.getCoding().add(new Coding()
+                .setCode(getInterpretationCodeAbbreviation(code))
+                .setDisplay(getInterpretationDisplay(code))
+                .setSystem(CODING_SYSTEM));
+
+            if (StringUtils.isNotEmpty(interpretationCode.getOriginalText())) {
+                interpretationCodeableConcept.setText(interpretationCode.getOriginalText());
+            } else if (StringUtils.isNotEmpty(interpretationCode.getDisplayName())) {
+                interpretationCodeableConcept.setText(interpretationCode.getDisplayName());
+            }
+
+            return interpretationCodeableConcept;
+        }
+
+        return null;
+    }
+
+    public static List<Observation.ObservationReferenceRangeComponent> getReferenceRange(
+        List<RCMRMT030101UK04ReferenceRange> referenceRangeList) {
+        var outputReferenceRanges = new ArrayList<Observation.ObservationReferenceRangeComponent>();
+
+        for (RCMRMT030101UK04ReferenceRange referenceRange : referenceRangeList) {
+            var referenceRangeComponent = new Observation.ObservationReferenceRangeComponent();
+            referenceRangeComponent.setText(referenceRange.getReferenceInterpretationRange().getText());
+
+            var quantity = QUANTITY_MAPPER.mapQuantity(referenceRange.getReferenceInterpretationRange().getValue());
+
+            var referenceInterpretationRange = referenceRange.getReferenceInterpretationRange();
+            if (referenceInterpretationRangeHasValue(referenceInterpretationRange)) {
+                if (referenceInterpretationRange.getValue().getLow() != null) {
```

Comment:
I think you should check if referenceInterpretationRange.getValue is not null as well